src/app/features/dialogs/cancel-transaction-dialog/hooks/use-stx-cancel-transaction.ts

```diff
@@ -0,0 +1,89 @@
+import { useCallback } from 'react';
+
+import { stxToMicroStx } from '@leather-wallet/utils';
+import { TransactionTypes } from '@stacks/connect';
+import * as yup from 'yup';
+
+import { useRefreshAllAccountData } from '@app/common/hooks/account/use-refresh-all-account-data';
+import {
+  type GenerateUnsignedTransactionOptions,
+  generateUnsignedTransaction,
+} from '@app/common/transactions/stacks/generate-unsigned-txs';
+import { getBurnAddress } from '@app/common/utils/get-burn-address';
+import { safelyFormatHexTxid } from '@app/common/utils/safe-handle-txid';
+import { stxFeeValidator } from '@app/common/validation/forms/fee-validators';
+import { useStacksBroadcastTransaction } from '@app/features/stacks-transaction-request/hooks/use-stacks-broadcast-transaction';
+import { useCurrentStxAvailableUnlockedBalance } from '@app/query/stacks/balance/account-balance.hooks';
+import { useCalculateStacksTxFees } from '@app/query/stacks/fees/fees.hooks';
+import { useCurrentStacksAccount } from '@app/store/accounts/blockchain/stacks/stacks-account.hooks';
+import { useCurrentStacksNetworkState } from '@app/store/networks/networks.hooks';
+import { useSubmittedTransactionsActions } from '@app/store/submitted-transactions/submitted-transactions.hooks';
+import { useRawDeserializedTxState, useRawTxIdState } from '@app/store/transactions/raw.hooks';
+
+import { useSelectedTx } from './use-selected-tx';
+
+export function useStxCancelTransaction() {
+  const [rawTxId, setRawTxId] = useRawTxIdState();
+  const tx = useSelectedTx();
+  const [, setTxId] = useRawTxIdState();
```

Comment:
I am going to block this PR from getting merged. So much of this code is just duplicated w/out evaluating if needed or even working correctly. I have had to refactor quite a bit of the increase fee dialog changes here: https://github.com/leather-wallet/extension/pull/5500

See here where I'm making this comment, there are two uses of `useRawTxIdState()` in this file to set the id. This component is a mess and caused and infinite loop and broke the wallet in the linked PR.